### DISCUSSION TITLE
EssentialsX Discord and Discord Link command role permissions not persisting through server restart

### DISCUSSION THREAD

**Discussion by coreyg142** [EssentialsX/Essentials#5431] [Q&A] [answered]
I am trying to add role overrides for the bot commands in Discord that restrict their visibility and usage to only certain roles. This is indicated by the "Has Overrides" tag in the image linked below.

https://i.imgur.com/Rvz2gUz.png

However, upon a server restart, these role overrides are wiped. I've linked a second image below showing the same screen but no "Has Overrides" tag, I did not clear them myself, all I did was restart the MC server. 

https://i.imgur.com/QZSCQlM.png

Is there any way to make these permissions persist through restarts?

**Comment by Chew** [comment]
The commands are removed when the server stops, I would assume by design, to make sure they aren't ran when the server is offline. I would honestly blame Discord for this blunder. The best we can do is not hide the commands on server stop.

**Comment by mdcfe** [comment]
Paging @JRoy - would this be a valid reason to persist commands across server reboot instead of clearing them on shutdown?

**Reply by JRoy** [reply, marked as answer]
Yes, when I overhaul the permissions system that's the plan
